arbeitszeit_flask/filters.py

```diff
@@ -0,0 +1,100 @@
+import os
+from typing import Callable, Dict
+
+from markupsafe import Markup
+
+ICON_PATH = "arbeitszeit_flask/templates/icons"
+
+
+def file_reader(file_path: str) -> str:
+    with open(file_path, "r") as file:
+        return file.read()
+
+
+def icon_filter(
+    icon_name: str,
+    reader: Callable[[str], str] = file_reader,
+    attrs: Dict[str, str] = {},
+) -> Markup:
+    """
+    This function inserts SVG icons into Flask templates with
+    customizable attributes.
+
+    Purpose:
+    --------
+    The `icon_filter` function is designed to dynamically include SVG
+    icons in Flask templates. It reads SVG files from a directory,
+    specified in `ICON_PATH`, injects additional HTML attributes into
+    the <svg> tag, and returns the SVG content safely wrapped in a
+    `Markup` object to prevent XSS attacks.
+
+    Usage (most cases):
+    -------------------
+    Assuming an SVG icon file named `example.html` exists in the `ICON_PATH`
+    directory, you can use the as `icon` registered `icon_filter` function
```

Comment:
OK, then this needs clarification. In Flask (the web framework), you can register filter functions, which can be used globally in any of the Flask (Jinja2) HTML template files. Therefore, in this PR, the filter function named `icon_filter` gets registered as a template filter named `icon` to transform a string from - for example - "user" (via) `{{ "user"|icon }}` into another string which is, in our case, an HTML snippet containing a svg-tag element derived from a file named "user.html" in the `arbeitszeit_flask/templates/icons` folder.

Reply:
In other words, to not have to write {{ "name"|icon_filter }} (hence the "icon_filter" part) in the template each time because, from the usage of this filter in the template, it's clear that it is a filter function. Therefore, the filter function "icon_filter" is registered as an "icon" and used as {{ "name"|icon }} in that template.

But why not call the original function "icon" in the first place? Because of context. If I import and use this function somewhere in the Python code, it would need to be clearer to the (human) reader that this Python is a template filter function. Therefore, I would name all filter functions in the Python code by convention `<name>_filter`.

Reply:
Thank you!  (So is a "filter function" actually what I would call a replacement function?)